spec/git-version-spec.cr

```diff
@@ -505,4 +505,61 @@ describe GitVersion do
       tmp.cleanup
     end
   end
-end
\ No newline at end of file
+
+  it "should properly manage prefixes" do
+    tmp = InTmp.new
+
+    begin
+      git = GitVersion::Git.new("dev", "master", tmp.@tmpdir, "v")
+
+      tmp.exec %(git init)
+      tmp.exec %(git checkout -b master)
+      tmp.exec %(git commit --no-gpg-sign --allow-empty -m "1")
+      tmp.exec %(git tag "v1.0.0")
+
+      version = git.get_version
+      version.should eq("v1.0.1")
+
+    ensure
+      tmp.cleanup
+    end
+  end
+
+  it "should properly switch from plain versioning to prefix versioning" do
+    tmp = InTmp.new
+
+    begin
+      git = GitVersion::Git.new("dev", "master", tmp.@tmpdir, "v")
+
+      tmp.exec %(git init)
+      tmp.exec %(git checkout -b master)
+      tmp.exec %(git commit --no-gpg-sign --allow-empty -m "1")
+      tmp.exec %(git tag "1.0.0")
+
+      version = git.get_version
+      version.should eq("v1.0.1")
```

Comment:
this should be `v1.0.0`
i.e.:
we need to check if https://github.com/codacy/git-version/pull/23/files#diff-8b0642b62d284c9602ddab77b3283422R89 has really stripped something.

Reply:
I don't get the reasoning behind this..
So if the semantically highest tag you have in the repository is `1.0.0` wouldn't you bump to the next patch version `1.0.1` with the prefix on it? `1.0.1`
Otherwise you have two versions `1.0.0` and `v1.0.0` and you don't know which one came before the other.. It sounds confusing to me.
Can you elaborate?

Reply:
Sure, since `git-version` is configured to read versions with a prefix it should exclusively read versions with the prefix, ignoring all the rest.

Basically `git tag "1.0.0"` should NOT be recognized as a version since it has not the correct format (in this case it's missing the trailing `v`)

And yes you will end up having two versions: `1.0.0` and `v1.0.0` but that's up to the user to be managed.

Reply:
So in that case it should create `0.0.1` because there are no other unprefixed tags. Right?

Reply:
correct

Reply:
Done ✅